### PR TITLE
feat(Progress stepper): Updated to use render prop for popover

### LINKED ISSUE #6658: ProgressStepper - investigate help popover using a ref callback
This would eliminate the need to set an additional reference property on the `ProgressStep` just for the `Popover`.

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #6658

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]
Looks good! Other than the below comment, I opened issue #7200 after noticing something when testing the preview build (unrelated to this PR/issue)

File: packages/react-core/src/components/ProgressStepper/ProgressStep.tsx [hunk 1]
```diff
@@ -22,10 +22,8 @@ export interface ProgressStepProps
   description?: string;
   /** ID of the title of the progress step. */
   titleId?: string;
-  /** Popover for a progress step. */
-  popover?: React.ReactNode;
-  /** @hide Forwarded reference to title container */
-  innerRef?: React.Ref<any>;
+  /** Forwards the stepRef to rendered function.  Use this prop to add a popover to the step.*/
+  render?: ({ stepRef }: { stepRef: React.RefObject<any> }) => React.ReactNode;
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ProgressStepper/ProgressStep.tsx:26]: Genuine question: why does `stepRef` need to be in an object? I pulled this branch locally and have `render` taking `stepRef` directly, and I *think* everything's still working properly.

**Comment by nicolethoen** [comment]
I'm noticing that the keyboard accessibility is still missing from the example with a popover. Did you have a chance to peek at that?

File: packages/react-core/src/components/ProgressStepper/ProgressStep.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-core/src/components/ProgressStepper/ProgressStep.tsx]: I was getting a lint error when I was using the innerRef we had before. I removed it so I no longer need an object.

File: packages/react-integration/demo-app-ts/src/components/demos/ProgressStepperDemo/ProgressStepperDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          )}
+        >
+          First step
+        </ProgressStep>
+        <ProgressStep
+          variant="danger"
+          id="popover-step2"
+          titleId="popover-step2-title"
+          aria-label="completed step with popover, step with danger"
+          popoveRender={({ stepRef }) => (
```
Comment (kmcfaul) [inline packages/react-integration/demo-app-ts/src/components/demos/ProgressStepperDemo/ProgressStepperDemo.tsx:34]: Missing 'r' here is why the build is failing atm.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**